Src/Witsml/Data/MudLog/WitsmlMudLogGeologyInterval.cs

```diff
@@ -0,0 +1,132 @@
+using System.Collections.Generic;
+using System.Xml.Serialization;
+
+using Witsml.Data.Measures;
+
+namespace Witsml.Data.MudLog
+{
+    public class WitsmlMudLogGeologyInterval : IWitsmlQueryType
+    {
+        [XmlAttribute("uid")]
+        public string Uid { get; set; }
+
+        [XmlElement("typeLithology")]
+        public string TypeLithology { get; set; }
+
+        [XmlElement("mdTop")]
+        public WitsmlIndex MdTop { get; set; }
+
+        [XmlElement("mdBottom")]
+        public WitsmlIndex MdBottom { get; set; }
+
+        [XmlElement("dTim")]
+        public string DTim { get; set; }
+
+        [XmlElement("tvdTop")]
+        public WitsmlMeasureWithDatum TvdTop { get; set; }
+
+        [XmlElement("tvdBase")]
+        public WitsmlMeasureWithDatum TvdBase { get; set; }
+
+        [XmlElement("ropAv")]
+        public Measure RopAv { get; set; }
+
+        [XmlElement("ropMn")]
+        public Measure RopMn { get; set; }
+
+        [XmlElement("ropMx")]
+        public Measure RopMx { get; set; }
+
+        [XmlElement("wobAv")]
+        public Measure WobAv { get; set; }
+
+        [XmlElement("tqAv")]
+        public Measure TqAv { get; set; }
+
+        [XmlElement("currentAv")]
+        public Measure CurrentAv { get; set; }
+
+        [XmlElement("rpmAv")]
+        public Measure RpmAv { get; set; }
+
+        [XmlElement("wtMudAv")]
+        public Measure WtMudAv { get; set; }
+
+        [XmlElement("ecdTdAv")]
+        public Measure EcdTdAv { get; set; }
+
+        [XmlElement("dxcAv")]
+        public string DxcAv { get; set; }
+
+        [XmlElement("lithology")]
+        public List<WitsmlMudLogLithology> Lithologies { get; set; }
+
+        [XmlElement("show")]
+        public WitsmlShow Show { get; set; }
+
+        [XmlElement("chromatograph")]
+        public WitsmlChromatograph Chromatograph { get; set; }
+
+        [XmlElement("mudGas")]
+        public WitsmlMudGas MudGas { get; set; }
+
+        [XmlElement("densBulk")]
+        public Measure DensBulk { get; set; }
+
+        [XmlElement("densShale")]
+        public Measure DensShale { get; set; }
+
+        [XmlElement("calcite")]
+        public Measure Calcite { get; set; }
+
+        [XmlElement("dolomite")]
+        public Measure Dolomite { get; set; }
+
+        [XmlElement("cec")]
+        public Measure Cec { get; set; }
+
+        [XmlElement("qft")]
+        public Measure Qft { get; set; }
+
+        [XmlElement("calcStab")]
+        public Measure CalcStab { get; set; }
+
+        [XmlElement("nameFormation")]
```

Comment:
`nameFormation` is flagged as deprecated and replaced by `lithostratigraphic with kind=formation`, should we indicate this in some way in the code?